skimage/data/generate_shapes.py

```diff
@@ -0,0 +1,316 @@
+        The size of the image into which this shape will be fit.
+    shape : ShapeProperties
+        The minimum and maximum size and color of the shape to fit.
+    random : np.random.RandomState
+        The random state to use for random sampling.
+
+    Raises
+    ------
+    ArithmeticError
+        When a shape cannot be fit into the image with the given starting
+        coordinates (x_0, y_0). This usually means the image dimensions are too
+        small or shape dimensions too large.
+
+    Returns
+    -------
+    mask : 3-D array
+        The shape mask that can be applied to an image.
+    label : Label
+        A tuple specifying the category of the shape, as well as its x1, x2, y1
+        and y2 bounding box coordinates.
+    """
+    available_width = min(image.ncols - point.column, shape.max_size)
+    if available_width < shape.min_size:
+        raise ArithmeticError('cannot fit shape to image')
+    available_height = min(image.nrows - point.row, shape.max_size)
+    if available_height < shape.min_size:
+        raise ArithmeticError('cannot fit shape to image')
+    # Pick random widths and heights.
+    r, c = random.randint(shape.min_size, available_width + 1, size=2)
+    mask = np.zeros((image.nrows, image.ncols, image.depth), dtype=np.uint8)
+    mask[point.row:point.row + r, point.column:point.column + c] = shape.color
+    label = Label('rectangle', point.column, point.column + c, point.row,
+                  point.row + r)
+
+    return mask, label
+
+
+def _generate_circle_mask(point, image, shape, random):
+    """Generate a mask for a filled circle shape.
+
+    The radius of the circle is generated randomly.
+
+    Parameters
+    ----------
+    point : Point
+        The row and column of the top left corner of the rectangle.
+    image : ImageShape
+        The size of the image into which this shape will be fit.
+    shape : ShapeProperties
+        The minimum and maximum size and color of the shape to fit.
+    random : np.random.RandomState
+        The random state to use for random sampling.
+
+    Raises
+    ------
+    ArithmeticError
+        When a shape cannot be fit into the image with the given starting
+        coordinates (x_0, y_0). This usually means the image dimensions are too
+        small or shape dimensions too large.
+
+    Returns
+    -------
+    mask : 3-D array
+        The shape mask that can be applied to an image.
+    label : Label
+        A tuple specifying the category of the shape, as well as its x1, x2, y1
+        and y2 bounding box coordinates.
+    """
+    if shape.min_size == 1 or shape.max_size == 1:
+        raise ValueError('size must be > 1 for circles')
+    min_radius = shape.min_size / 2
+    max_radius = shape.max_size / 2
+    left = point.column
+    right = image.ncols - point.column
+    top = point.row
+    bottom = image.nrows - point.row
+    available_radius = min(left, right, top, bottom, max_radius)
+    if available_radius < min_radius:
```

Comment:
Much of the shape checking is already performed by skimage.draw.  Can't we re-use that functionality?

Reply:
A lot of this is actually so that `randint` doesn't throw for (low >= high)